tests/benchmark-models/test_petab_model.py

```diff
@@ -0,0 +1,139 @@
+                        help='Plot measurement and simulation results')
+
+    # PEtab problem
+    parser.add_argument('-y', '--yaml', dest='yaml_file_name',
+                        required=True,
+                        help='PEtab YAML problem filename')
+
+    # Corresponding AMICI model
+    parser.add_argument('-m', '--model-name', dest='model_name',
+                        help='Name of the AMICI module of the model to '
+                             'simulate.', required=True)
+    parser.add_argument('-d', '--model-dir', dest='model_directory',
+                        help='Directory containing the AMICI module of the '
+                             'model to simulate. Required if model is not '
+                             'in python path.')
+
+    parser.add_argument('-o', '--simulation-file', dest='simulation_file',
+                        help='File to write simulation result to, in PEtab'
+                        'measurement table format.')
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    """Simulate the model specified on the command line"""
+
+    args = parse_cli_args()
+
+    if args.verbose:
+        logger.setLevel(logging.DEBUG)
+
+    logger.info(f"Simulating '{args.model_name}' "
+                f"({args.model_directory}) using PEtab data from "
+                f"{args.yaml_file_name}")
+
+    # load PEtab files
+    problem = petab.Problem.from_yaml(args.yaml_file_name)
+
+    # load model
+    if args.model_directory:
+        sys.path.insert(0, args.model_directory)
+    model_module = importlib.import_module(args.model_name)
+    amici_model = model_module.getModel()
+
+    res = simulate_petab(
+        petab_problem=problem, amici_model=amici_model,
+        log_level=logging.DEBUG)
+    rdatas = res[RDATAS]
+    llh = res[LLH]
+
+    # create simulation PEtab table
+    sim_df = rdatas_to_measurement_df(rdatas=rdatas, model=amici_model,
+                                      measurement_df=problem.measurement_df)
+    sim_df.rename(columns={petab.MEASUREMENT: petab.SIMULATION}, inplace=True)
+
+    if args.simulation_file:
+        sim_df.to_csv(index=False, sep="\t")
+
+    if args.plot:
+        try:
+            # visualize fit
+            ax = plot_petab_problem(petab_problem=problem, sim_data=sim_df)
+
+            # save figure
+            fig_path = os.path.join(args.model_directory,
+                                    args.model_name + "_vis.png")
+            logger.info(f"Saving figure to {fig_path}")
+            ax.flatten()[0].get_figure().savefig(fig_path, dpi=150)
+
+        except NotImplementedError:
+            pass
+
+    if args.check:
+        references_yaml = os.path.join(os.path.dirname(__file__),
+                                       "benchmark_models.yaml")
+        with open(references_yaml) as f:
+            refs = yaml.full_load(f)
+
+        try:
+            ref_llh = refs[args.model_name]["llh"]
```

Comment:
this would be a good spot for _in the future_ also checking if the computed gradients match with finite difference gradients.